metricshub-agent/src/main/java/org/metricshub/web/config/SecurityConfig.java

```diff
@@ -53,8 +76,50 @@ public SecurityConfig(ApiKeyAuthFilter apiKeyAuthFilter) {
 	public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
 		return http
 			.csrf(csrf -> csrf.disable())
+			.sessionManagement(sm -> sm.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
+			// Order matters: API key first, JWT after it.
 			.addFilterBefore(apiKeyAuthFilter, UsernamePasswordAuthenticationFilter.class)
-			.authorizeHttpRequests(authz -> authz.anyRequest().authenticated())
+			.addFilterAfter(new JwtAuthFilter(jwtComponent, userService), ApiKeyAuthFilter.class)
+			.authorizeHttpRequests(authz -> authz.requestMatchers("/auth").permitAll().anyRequest().authenticated())
+			.exceptionHandling(ex ->
+				ex.authenticationEntryPoint(jsonAuthEntryPoint()).accessDeniedHandler(jsonForbiddenHandler())
+			)
 			.build();
 	}
+
+	/**
+	 * JSON 401 for unauthenticated.
+	 *
+	 * @return the authentication entry point
+	 */
+	@Bean
+	public AuthenticationEntryPoint jsonAuthEntryPoint() {
+		return (HttpServletRequest request, HttpServletResponse response, AuthenticationException authException) -> {
+			SecurityHelper.writeUnauthorizedResponse(response);
+		};
+	}
+
+	/**
+	 * JSON 403 for forbidden.
+	 *
+	 * @return the access denied handler
+	 */
+	@Bean
+	public AccessDeniedHandler jsonForbiddenHandler() {
+		return (HttpServletRequest request, HttpServletResponse response, AccessDeniedException accessDeniedException) -> {
+			SecurityHelper.writeForbiddenResponse(response);
+		};
+	}
+
+	/**
+	 * Configures web security to ignore specific URL patterns, allowing unrestricted access to them.
+	 *
+	 * @return the configured WebSecurityCustomizer
+	 */
+	@Bean
+	public WebSecurityCustomizer webSecurityCustomizer() {
+		final var allowedPatterns = new String[] { "/", "/assets/**", "/index.html" };

```

Comment:
It will probably be easier to maintain to only secure the API (`/api/*`) and allow access to all other resources.